byway/src/main/webapp/base.js

```diff
@@ -92,3 +93,51 @@ function findPlace(placeId, placesService) {
   return result;
 }
 
+/**
+ * Class used to throw error in getDirections() with both message and status
+ */
+class MapStatusError extends Error {
+    constructor(message, status) {
+        super(message);
+        this.name = 'MapStatusError';
+        this.status = status;
+    }
+}
+
+/**
+ * Makes a request to the Directions API to return information about the route
+ * @param {DirectionsService} directionsService object that communicates with the GMaps API service
+ * @param {directionsRenderer} directionsRenderer object that renders display results on the map
+ * @param {String} start placeId as string
+ * @param {String} end placeId as string
+ * @param {Array} [waypoints] array of waypoint objects
+ * @return {Promise} result Directions route object with fields route, legs, etc.
+ */
+function getRouteForTrip(directionsService, directionsRenderer, start, end, waypoints) {
+   let request = {
+    origin:  {placeId : start},
+    destination: {placeId : end},
+    waypoints, 
+    travelMode: 'DRIVING',
+    optimizeWaypoints: true
+  };
+  const result = new Promise((resolve,reject) => {
+    directionsService.route(request, (result, status) => {
+      if (status == 'OK') {
+        directionsRenderer.setDirections(result);
+        resolve(result);
+      }
+      else {
+        reject(new MapStatusError("Could not calculate route from request.", status));
+      }
+    });
+  });
+  return result;
+}
+
+/** Displays error message to the user if a request fails.*/
+function showErrorMessage(errorMessage){
+  let msgContainer = document.getElementById("message-container");
```

Comment:
Since this is in the `base.js`, may not be guaranteed to work for all documents when using this script tag. This means all pages need a tag with that specific id. I think it'd be better to pass in the HTML element as a parameter and set it that way, rather than require all pages to have an element with that specific id.

Reply:
I spoke to Rena about also adding errors to the pages she worked on, and suggested message-container be a common element in the base.css. So we do have the message-container elements coming up in another PR, but IDs are supposed to be unique so I'm not sure if this the best way to do it either. Will wait for Zach or Landon to comment.